cpu/native/mtdi/native_mtdi.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+
+#include "mtdi.h"
+#include "native_mtdi.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static const char _fname[] = "MEMORY.bin";
+
+#ifndef NATIVE_MTDI_PAGE_SIZE
+#define NATIVE_MTDI_PAGE_SIZE     256
+#endif
+#ifndef NATIVE_MTDI_PAGE_NUM
+#define NATIVE_MTDI_PAGE_NUM      32768
+#endif
+#ifndef NATIVE_MTDI_SECTOR_SIZE
+#define NATIVE_MTDI_SECTOR_SIZE   4096
+#endif
+#ifndef NATIVE_MTDI_SECTOR_NUM
+#define NATIVE_MTDI_SECTOR_NUM    2048
+#endif
+#ifndef NATIVE_MTDI_FLASH_SIZE
+#define NATIVE_MTDI_FLASH_SIZE    (NATIVE_MTDI_SECTOR_SIZE * NATIVE_MTDI_SECTOR_NUM)
+#endif
+
+static mtdi_sta_t _init(mtdi_dev_t *mtdi_dev)
+{
+    (void) mtdi_dev;
+
+    FILE *f = fopen(_fname, "r");
```

Comment:
Please review the system call usage pattern of native system calls e.g. in https://github.com/RIOT-OS/RIOT/pull/2239.